### PR TITLE
A few advanced appearance attrs

### PR DESCRIPTION
**Pull request by benrudolph**
@wpride added some advanced appearance attributes to web apps
screen shot 2017-01-25 at 1 17 02 pm
screen shot 2017-01-25 at 1 32 55 pm

Adds the following:

- Hint text
- Help text
- Dropdown instead of single select (this doesn't currently work because the backend doesn't send the proper appearance property. once that is fixed though this will work and it doesn't hurt to merge)

cc: @emord 
design: @biyeun / @orangejenny

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 1]
```diff
@@ -303,6 +303,27 @@ SingleSelectEntry.prototype.onPreProcess = function(newValue) {
     }
 };
 
+function DropdownEntry(question, options) {
+    var self = this;
+    EntrySingleAnswer.call(this, question, options);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: 'EntrySingleAnswer' is not defined. (no-undef)

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 2]
```diff
@@ -303,6 +303,27 @@ SingleSelectEntry.prototype.onPreProcess = function(newValue) {
...
+    self.templateType = 'dropdown';
+    self.options = _.map(question.choices(), function(choice, idx) {
+        return { text: choice, idx: idx + 1 };
+    });
+}
+DropdownEntry.prototype = Object.create(EntrySingleAnswer.prototype);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: 'EntrySingleAnswer' is not defined. (no-undef)

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 3]
```diff
@@ -303,6 +303,27 @@ SingleSelectEntry.prototype.onPreProcess = function(newValue) {
...
+DropdownEntry.prototype.constructor = EntrySingleAnswer;
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: 'EntrySingleAnswer' is not defined. (no-undef)

**Comment by wpride** [comment]
very sexy benjo kazooie

**Review by wpride — APPROVED** [review]

**Comment by orangejenny** [comment]
nice, I like how well it mimics the phone UI

**Pull request merged**